update_twiki.py

```diff
@@ -0,0 +1,109 @@
+                  target_num_events real NOT NULL   )''')
+
+    # Clear the table
+    c.execute('DELETE FROM `twiki_samples`')
+    conn.commit()
+    conn.close()
+
+def insert_update(twiki_request, campaign, cross_section, frac_neg_wgts, target_num_events):
+    conn = sqlite3.connect('twiki.db')
+    c = conn.cursor()
+    logger.info('Inserting %s (%s) %s %s %s', twiki_request['dataset_name'], twiki_request['prepid'], cross_section, frac_neg_wgts, target_num_events)
+    c.execute('INSERT INTO twiki_samples VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)',
+              [twiki_request['prepid'],
+               twiki_request['dataset_name'],
+               twiki_request['extension'],
+               twiki_request['total_events'],
+               campaign,
+               twiki_request['prepid'].split('-')[0],
+               cross_section,
+               frac_neg_wgts,
+               target_num_events])
+
+    conn.commit()
+    conn.close()
+
+def operations():
+    query_ul18 = 'member_of_campaign=RunIISummer19UL18MiniAOD'
+    query_ul16 = 'member_of_campaign=RunIISummer19UL16MiniAOD'
+    query_ul17 = 'member_of_campaign=RunIISummer19UL17MiniAOD'
+    requests_ul17 = mcm.get('requests', query=query_ul17)
+    requests_ul18 = mcm.get('requests', query=query_ul18)
+    requests_ul16 = mcm.get('requests', query=query_ul16)
+    ul17_dataset_names = {x['dataset_name'] for x in requests_ul17}
+    ul16_dataset_names = {x['dataset_name'] for x in requests_ul16}
+    ul18_dataset_names = {x['dataset_name'] for x in requests_ul18}
+    missing_ul18 = ul17_dataset_names - ul18_dataset_names
+    missing_ul16 = ul17_dataset_names - ul16_dataset_names
+
+    # Get all needed requests
+    total_events_threshold = 0#20000000
+    for twiki_request in requests_ul17:
+        if twiki_request['total_events'] > total_events_threshold:
+            campaign = 'RunIISummer19UL17'
+            if twiki_request['dataset_name'] in missing_ul18:
+                campaign = 'RunIISummer19UL18'
+            if twiki_request['dataset_name'] in missing_ul16:
+                campaign = 'RunIISummer19UL16'
+            if campaign == 'RunIISummer19UL17':
+                continue
+            #Getting the cross_section value from xsdb
+            query = {'DAS': twiki_request['dataset_name']}
+            search_rslt = xsdb_request.simple_search_to_dict(query)
+            cross_section = -1
+            frac_neg_wgts = 0
+            target_num_events = -1
+            if search_rslt:
+                search_rslt_ = search_rslt[-1]
+                cross_section = float(search_rslt_[u'cross_section'])
+                frac_neg_wgts = float(search_rslt_[u'fraction_negative_weight'])
+            else:
+                try:
+                    cross_section = float(twiki_request['generator_parameters'][-1][u'cross_section'])
+                    frac_neg_wgts = float(twiki_request['generator_parameters'][-1][u'negative_weights_fraction'])
+                except:
```

Comment:
Even though pylint complains about not specifying exception type, it is also a good idea to print whatever exceptions this `except` catches.
```
except Exception as ex:
    logger.error(ex)
    logger.error(twiki_request['generator_parameters'])
```

```
W: 95,16: No exception type(s) specified (bare-except)
```